### ISSUE TITLE
Autosave is not retried after failing once

### ISSUE THREAD

**Issue by jsnajdr** [WordPress/gutenberg#54969]
While working on an autosave issue I noticed another one that would be nice to fix:
1. Start editing an existing post, draft or published.
2. Change some content so that autosave is requested.
3. In devtools, change network to Offline before the autosave request is triggered.
4. The autosave request will fail. Notice "Update failed, you are probably offline" appears.
5. Go back online.

**Expected result** is that the editor sooner or later retries the failed autosave, and I don't need to make any extra edits to trigger it.

**Actual result** is that the autosave is never triggered again until you make further changes. The existing changes don't get autosaved.

@flootr might be interested in looking at this.

**Comment by Mamaduka** [comment]
I thought `apiFetch` would retry failed requests automatically, or that's only for expired nonces?

Sidenote: We also want to implement a `maxRetries` limit. See #42400.

**Comment by tyxla** [comment]
I wonder if #13509 is still valid. If it is, it could be one of the potential reasons for save failures and we should likely handle it in a better way.

**Comment by Mamaduka** [comment]
@tyxla, yes, that's still a valid issue. Here are my detailed findings - https://github.com/WordPress/gutenberg/issues/13509#issuecomment-1210325129.

**Comment by Mamaduka** [comment]
I created a PR (#67812) to resolve the infinite loop request issue when the session token is missing, but it doesn't seem to fix the problem, at least not in my tests.

I think the problem could be in the autosave monitor component and how autosaving is rescheduled.
